common/jsonManager_test.go

```diff
@@ -0,0 +1,315 @@
+	t.Parallel()
+
+	j := helperCreateJSON(t, testJSONData) // nolint:varnamelen
+
+	type inType struct {
+		key      string
+		optional bool
+	}
+
+	tests := []struct {
+		name        string
+		input       inType
+		expected    *string
+		expectedErr error
+		withErr     bool
+	}{
+		{
+			name: "Missing string",
+			input: inType{
+				key: "surname",
+			},
+			withErr: true,
+		},
+		{
+			name: "Missing optional string",
+			input: inType{
+				key:      "surname",
+				optional: true,
+			},
+			expected: nil,
+		},
+		{
+			name: "Incorrect data type",
+			input: inType{
+				key: "count",
+			},
+			expectedErr: ErrNotString,
+		},
+		{
+			name: "Key is found but Null",
+			input: inType{
+				key: "metadata",
+			},
+			expectedErr: ErrNullJSON,
+		},
+		{
+			name: "Valid string",
+			input: inType{
+				key: "text",
+			},
+			expected: ptrString("Hello World"),
+		},
+	}
+
+	for _, tt := range tests {
+		// nolint:varnamelen
+		tt := tt // rebind, omit loop side effects for parallel goroutine
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			output, err := JSONManager.GetString(j, tt.input.key, tt.input.optional)
+			assertJSONManagerOutput(t, tt.name, tt.expected, tt.expectedErr, tt.withErr, output, err)
+		})
+	}
+}
+
+func TestJsonManager_GetArr(t *testing.T) {
+	t.Parallel()
+
+	j := helperCreateJSON(t, testJSONData) // nolint:varnamelen
+
+	tests := []struct {
+		name         string
+		input        string
+		expectedSize int
+		expectedErr  error
+		withErr      bool
+	}{
+		{
+			name:         "Missing array",
+			input:        "queue",
+			expectedSize: 0,
+			withErr:      true,
+		},
+		{
+			name:        "Key is found but Null",
+			input:       "metadata",
+			expectedErr: ErrNotArray,
+		},
+		{
+			name:         "Valid array",
+			input:        "list",
+			expectedSize: 3,
+		},
+		{
+			name:         "Empty array",
+			input:        "arr",
+			expectedSize: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		// nolint:varnamelen
+		tt := tt // rebind, omit loop side effects for parallel goroutine
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			output, err := JSONManager.GetArr(j, tt.input)
+			if tt.withErr {
+				if err == nil {
+					t.Fatalf("%s: expected error while none received", tt.name)
+				}
+			} else {
+				if !errors.Is(err, tt.expectedErr) {
+					t.Fatalf("%s: expected: (%v), got: (%v)", tt.name, tt.expectedErr, err)
+				}
+
+				outputSize := len(output)
+				if !reflect.DeepEqual(outputSize, tt.expectedSize) {
+					t.Fatalf("%s: expected: (%v), got: (%v)", tt.name, tt.expectedSize, outputSize)
+				}
+			}
+		})
+	}
+}
+
+func TestJsonManager_GetNestedNode(t *testing.T) {
+	t.Parallel()
+
+	j := helperCreateJSON(t, testJSONData) // nolint:varnamelen
+
+	tests := []struct {
+		name        string
+		input       string
+		expectedErr error
+		withErr     bool
+	}{
+		{
+			name:    "Missing nested object",
+			input:   "payload.random",
```

Comment:
Why don't you make the input an array, just like what `GetNestedNode` expects? Then you don't have to parse it into an array on line 263.